miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppSdkConfig.kt

```diff
@@ -3,34 +3,44 @@ package com.rakuten.tech.mobile.miniapp
 /**
  * This represents the configuration settings for the Mini App SDK.
  * @property baseUrl Base URL used for retrieving a Mini App.
- * @property rasAppId App ID for the Platform API.
+ * @property rasProjectId Project ID for the Platform API.
+ * @property rasAppId App ID for the Platform API (Deprecated).
  * @property subscriptionKey Subscription Key for the Platform API.
  * @property hostAppVersionId Version of the host app, used to determine feature compatibility for Mini App.
  * @property hostAppUserAgentInfo User Agent information from Host App.
  * @property isTestMode Whether the sdk is making use of the Test API Endpoints for under "Testing" mini apps.
  */
 data class MiniAppSdkConfig(
     val baseUrl: String,
-    val rasAppId: String,
+    var rasProjectId: String,
+    @Deprecated("Use rasProjectId") var rasAppId: String,
     val subscriptionKey: String,
     val hostAppVersionId: String,
     val hostAppUserAgentInfo: String,
     val isTestMode: Boolean
 ) {
-    internal val key = "$baseUrl-$isTestMode-$rasAppId-$subscriptionKey-$hostAppVersionId"
+    internal var projectOrAppId = ""
 
     init {
         when {
             !isBaseUrlValid(baseUrl) ->
                 throw sdkExceptionForInvalidArguments("MiniAppSdkConfig with invalid baseUrl")
-            rasAppId.isBlank() ->
-                throw sdkExceptionForInvalidArguments("MiniAppSdkConfig with invalid rasAppId")
             subscriptionKey.isBlank() ->
                 throw sdkExceptionForInvalidArguments("MiniAppSdkConfig with invalid subscriptionKey")
             hostAppVersionId.isBlank() ->
                 throw sdkExceptionForInvalidArguments("MiniAppSdkConfig with invalid hostAppVersionId")
+            rasProjectId.isBlank() && rasAppId.isBlank() ->
```

Comment:
See my suggestion above - in that case you wouldn't need to handle `rasAppId` and `rasProjectId` separately.

Reply:
I have updated this part.